include/unifex/sender_concepts.hpp

```diff
@@ -402,9 +402,27 @@ struct single_value_type<> {
   using type = void;
 };
 
+template <typename... Overloads>
+struct single_overload {};
+
+template <typename Overload>
+struct single_overload<Overload> {
+  using type = Overload;
+};
+
+template <>
+struct single_overload<> {
+private:
+  struct dummy {
+    using type = void;
+  };
+public:
+  using type = dummy;
+};
```

Comment:
I initially had `struct type` but I can't name the nested `::type` type-alias then because `type` is reserved for the constructor name.
I'll rename to something like `impl`.